### PR TITLE
Flink: switch to FileScanTaskParser for JSON serialization of IcebergSourceSplit

### PR DESCRIPTION
**Pull request by stevenzwu**

### CODE REVIEW

**Review by pvary — APPROVED** [review]

File: flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java [hunk 1]
```diff
@@ -120,4 +131,56 @@ static IcebergSourceSplit deserializeV1(byte[] serialized) throws IOException {
...
     }
   }
+
+  byte[] serializeV2() throws IOException {
+    if (serializedBytesCache == null) {
+      DataOutputSerializer out = SERIALIZER_CACHE.get();
+      Collection<FileScanTask> fileScanTasks = task.tasks();
+      Preconditions.checkArgument(
+          fileOffset >= 0 && fileOffset < fileScanTasks.size(),
+          "Invalid file offset: %s. Should within the range of [0, %s)",
```
Comment (nastra) [inline flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java:141]:
```suggestion
          "Invalid file offset: %s. Should be within the range of [0, %s)",
```

File: flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java [hunk 2]
```diff
@@ -120,4 +131,56 @@ static IcebergSourceSplit deserializeV1(byte[] serialized) throws IOException {
...
+      int writtenTasks = 0;
+      while (iter.hasNext()) {
+        String taskJson = FileScanTaskParser.toJson(iter.next());
+        out.writeUTF(taskJson);
+        writtenTasks += 1;
+      }
+
+      Preconditions.checkArgument(
+          writtenTasks == fileScanTasks.size(),
+          "Invalid combined scan task: collection size = %s, iterator size = %s",
```
Comment (nastra) [inline flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java:159]: I'm not sure it's clear to a reader what the difference between `collection size` and `iterator size` is

**Review by nastra — APPROVED** [review]
LGTM, but I left a comment around error msg clarity

File: flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java (hunk 2 above)
Comment (stevenzwu) [reply flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java]: totally agree. removed this unnecessary check and simplified the loop.

**Comment by stevenzwu** [comment]
issue #1698

**Comment by stevenzwu** [comment]
thanks @nastra and @pvary for the review

**Pull request merged**